worker/worker.go

```diff
@@ -386,21 +386,27 @@ func (w *worker) withHosts(ctx context.Context, contracts []api.ContractMetadata
 	}
 	done := make(chan struct{})
 	go func() {
+		// apply a sane (pessimistic) timeout, ensuring unlocking the contract
+		// or force closing the session does not deadlock and keep this
+		// goroutine around forever
+		ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
```

Comment:
5 minutes seems ok for one host, but we're dealing with multiple hosts here. This timeout should be really extreme, like an hour maybe.